lib/phoenix/pubsub/local_supervisor.ex

```diff
@@ -0,0 +1,46 @@
+defmodule Phoenix.PubSub.LocalSupervisor do
+  use Supervisor
+  alias Phoenix.PubSub.Local
+
+  @moduledoc """
+  Local PubSub server supervisor.
+
+  Used by PubSub adapters to handle "local" subscriptions.
+  Defines an ets dispatch table for routing subcription
+  requests. Extendable by PubSub adapters by providing
+  a list of `dispatch_rules` to extend the dispatch table.
+
+  See `Phoenix.PubSub.PG2` for example usage.
+  """
+
+  def start_link(server, pool_size, dispatch_rules) do
+    Supervisor.start_link(__MODULE__, [server, pool_size, dispatch_rules])
+  end
+
+  @doc false
+  def init([server, pool_size, dispatch_rules]) do
+    # Define a dispatch table so we don't have to go through
+    # a bottleneck to get the instruction to perform.
+    ^server = :ets.new(server, [:set, :named_table, read_concurrency: true])
+    true = :ets.insert(server, {:subscribe, Phoenix.PubSub.Local, [server, pool_size]})
+    true = :ets.insert(server, {:unsubscribe, Phoenix.PubSub.Local, [server, pool_size]})
+    for rule <- dispatch_rules do
```

Comment:
Actually, you don't need the loop `:ets.insert` accepts many items I think (or there is a variant that does).